src/player.cpp

```diff
@@ -11602,14 +11602,22 @@ long player::get_memorized_symbol( const tripoint &p ) const
 
 size_t player::max_memorized_tiles() const
 {
-    if( has_active_bionic( bio_memory ) ) {
-        return SEEX * SEEY * 20000; // 5000 overmap tiles
-    } else if( has_trait( trait_FORGETFUL ) ) {
-        return SEEX * SEEY * 200; // 50 overmap tiles
-    } else if( has_trait( trait_GOODMEMORY ) ) {
-        return SEEX * SEEY * 800; // 200 overmap tiles
+    // Only check traits once a turn since this is called a huge number of times.
+    static time_point current_turn = calendar::before_time_starts;
+    static size_t current_max = 0;
```

Comment:
I feel that it's unnecessary to make these static, and they would more logically be member variables.  For example, when you save a game and load a different one, it's still using the state from the previous game.  Which could (if you got very unlucky) cause that new game's player to forget a bunch of map memory.

Reply:
Ugh you're right, there is a tiny collision risk.